src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -109,16 +118,32 @@
 
     private final long genomeSize;
 
+    private final int minimumMappingQuality;
+    private final int minimumBaseQuality;
+    private final boolean clipOverlappingReads;
+    private boolean noSideEffects;
+
     //A map of coverage by target in which coverage is reset every read, this is done
     //so that we can calculate overlap for a read once and the resulting coverage is
     //than added to the cumulative coverage of every collector that collects
     //information on that read
-    private Map<Interval, Coverage> coverageByTargetForRead;
-    private Coverage [] cov;
+    private final Map<Interval, Coverage> coverageByTargetForRead;
+    private final Coverage [] cov;
 
     //Converts a targetMetric into a more specific metric of METRIC_TYPE
     public abstract METRIC_TYPE convertMetric(final TargetMetrics targetMetrics);
 
+
+    /**
+     * In the case of ignoring bases in overlapping reads from the same template,
+     * we can choose whether to modify the existing record, or modify a clone.  The
+     * latter may significantly slow down the performance, but also has no side effects.
+     * @param value the boolean value to set.
```

Comment:
I'll try to rephrase in my next commit, but if we pass `false` to `setNoSideEffects`, then it implies we do not want any side-effects, so there really isn't anything to expand upon other than "there will be none".